include/circt/Dialect/RTL/RTLCombinatorial.td

```diff
@@ -267,3 +272,13 @@ def MuxOp : RTLOp<"mux",
     }]>
   ];
 }
+
+def ReinterpretCastOp : RTLOp<"reinterpret_cast", [NoSideEffect]> {
+  let summary = "Change types without any conversion";
+
+  let arguments = (ins AnyType:$arg);
+  let results = (outs AnyType:$out);
```

Comment:
I don't understand how this can be used without a model for what is allowed.  What is the usecase for this?  For example, this sounds like it would allow i32 to f32, but also f32 to i64 or f64 to f32.  

These sorts of cases need to be defined as part of the semantics of the op (or forbidden).

Reply:
The use case (for both of these additions) is data serialization. I've got a blob of untyped data coming in as an `array<8192xi1>` or `array<1024xi8>` or pretty much any array of ints. I need to slice and dice them according to some schema (implicitly or explicitly defined), then cast the untyped data to some type.

The intention was to only support casting data of equivalent bit widths.

Reply:
And the other way around -> typed data to untyped.

Reply:
And you're right: this is way too broad.